test/test_binary_ufuncs.py

```diff
@@ -1359,72 +1359,79 @@ def test_rdiv(self, device, dtype):
         self.assertEqual(y, z, exact_dtype=False)
 
     @dtypes(*torch.testing.get_all_fp_dtypes(include_bfloat16=False))
-    def test_fmod_by_zero_float(self, device, dtype):
-        # check floating-point tensor fmod to zero is nan on both CPU and GPU
-        x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
-        zero = torch.zeros_like(x)
-        self.assertTrue(torch.all(x.fmod(0.0).isnan()))
-        self.assertTrue(torch.all(x.fmod(zero).isnan()))
+    def test_fmod_remainder_by_zero_float(self, device, dtype):
+        fn_list = (torch.fmod, torch.remainder)
+        for fn in fn_list:
+            # check floating-point tensor fmod/remainder to zero is nan on both CPU and GPU
+            x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
+            zero = torch.zeros_like(x)
+            self.assertTrue(torch.all(fn(x, 0.0).isnan()))
+            self.assertTrue(torch.all(fn(x, zero).isnan()))
 
     @onlyOnCPUAndCUDA  # Check Issue https://github.com/pytorch/pytorch/issues/48130
     @skipCUDAIfRocm  # Error happens on both ROCM and XLA
     @dtypes(*torch.testing.get_all_int_dtypes())
-    def test_fmod_by_zero_integral(self, device, dtype):
-        # check integral tensor fmod to zero
-        x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
-        zero = torch.zeros_like(x)
-        # RuntimeError on CPU
-        if self.device_type == 'cpu':
-            with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
-                x.fmod(zero)
-        # Different value for different dtype on CUDA:
-        # Due to it's an undefined behavior, CUDA returns a pattern of all 1s
-        # for integral dividend (other than int64) divided by zero. For int64,
-        # CUDA returns all 1s for negative dividend, half 1s for positive dividend.
-        # uint8: 0xff -> 255
-        # int32: 0xffffffff -> -1
-        else:
-            if dtype == torch.int64:
-                self.assertEqual(x.fmod(zero) == 4294967295, x >= 0)
-                self.assertEqual(x.fmod(zero) == -1, x < 0)
+    def test_fmod_remainder_by_zero_integral(self, device, dtype):
+        fn_list = (torch.fmod, torch.remainder)
+        for fn in fn_list:
+            # check integral tensor fmod/remainder to zero
+            x = make_tensor((10, 10), device=device, dtype=dtype, low=-9, high=9)
+            zero = torch.zeros_like(x)
+            # RuntimeError on CPU
+            if device == 'cpu':
+                with self.assertRaisesRegex(RuntimeError, "ZeroDivisionError"):
+                    fn(x, zero)
+            # Different value for different dtype on CUDA:
```

Comment:
This note is very helpful. In a future PR we should probably update the documentation for fmod and remainder to include this, too.